devchat.go

```diff
@@ -45,19 +46,66 @@ var (
 	rtm       *slack.RTM
 	client    = loadTwitterClient()
 
-	red      = color.New(color.FgHiRed)
-	green    = color.New(color.FgHiGreen)
-	cyan     = color.New(color.FgHiCyan)
-	magenta  = color.New(color.FgHiMagenta)
-	yellow   = color.New(color.FgHiYellow)
-	blue     = color.New(color.FgHiBlue)
-	black    = color.New(color.FgBlack)
-	white    = color.New(color.FgHiWhite)
-	colorArr = []*color.Color{yellow, cyan, magenta, green, white, blue, red}
-
+	red         = color.New(color.FgHiRed)
+	green       = color.New(color.FgHiGreen)
+	cyan        = color.New(color.FgHiCyan)
+	magenta     = color.New(color.FgHiMagenta)
+	yellow      = color.New(color.FgHiYellow)
+	blue        = color.New(color.FgHiBlue)
+	black       = color.New(color.FgBlack)
+	white       = color.New(color.FgHiWhite)
 	devbot      = "" // initialized in main
 	startupTime = time.Now()
 
+	colorstyles = []colorstyle{
+		{"white", buildStyleApplicator(*white)},
+		{"red", buildStyleApplicator(*red)},
+		{"green", buildStyleApplicator(*green)},
+		{"cyan", buildStyleApplicator(*cyan)},
+		{"magenta", buildStyleApplicator(*magenta)},
+		{"yellow", buildStyleApplicator(*yellow)},
+		{"blue", buildStyleApplicator(*blue)},
+		{"black", buildStyleApplicator(*black)},
+		{"easter", buildStyleApplicator(*color.New(color.BgMagenta, color.FgHiYellow))},
+		{"baby", buildStyleApplicator(*color.New(color.BgBlue, color.FgHiMagenta))},
+		{"hacker", buildStyleApplicator(*color.New(color.FgHiGreen, color.BgBlack))},
+		//TODO: find a way to support special colors which rely on the user object
+		// {
+		// 	"l33t",
+		// 	func(a string, u user) string {
+		//		// TODO: this will lead to catastrophic failure with other special colors!
+		// 		c := *u.getColor().Add(color.BgHiBlack)
+		// 		return c.Sprint(stripansi.Strip(a))
+		// 	},
+		// },
+		// Same with "whiten", *u.getColor().Add(color.BgWhite)
+		{
+			"rainbow",
+			func(a string) string {
+				var rainbow = []*color.Color{
+					red,
+					magenta,
+					blue,
+					cyan,
+					yellow,
+					green,
+					white,
+				}
+
+				var stripped = stripansi.Strip(a)
+				var buf = ""
+				colorOffset := rand.Intn(len(rainbow) - 1)
```

Comment:
```suggestion
				colorOffset := rand.Intn(len(rainbow))
```
rand.Intn always returns a number less than the argument (`[0,n)`)